src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -407,8 +415,16 @@ MetadataCreateCommands(void)
 	/* construct the foreign key constraints after all tables are created */
 	foreach_ptr(cacheEntry, propagatedTableList)
 	{
+		Oid relationId = cacheEntry->relationId;
+
+		if (IsTableOwnedByExtension(relationId))
+		{
+			/* skip foreign key creation when the Citus table is owned by an extension */
+			continue;
```

Comment:
Interesting, now codecov shows us which lines are not covered:

![image](https://user-images.githubusercontent.com/17179875/92710791-c2de9b00-f360-11ea-986f-5f7f4fffba86.png)

Reply:
That's going to be quite intrusive in some of our code paths. Anyway, should be covered now.